tick/stateful/stateful_expr.go

```diff
@@ -0,0 +1,95 @@
+package stateful
+
+import (
+	"fmt"
+
+	"github.com/influxdata/kapacitor/tick"
+)
+
+// StatefulExpression is interface that describe expression with state and
+// it's evaluation.
+type StatefulExpression interface {
+	Reset()
+
+	EvalFloat(scope *tick.Scope) (float64, error)
+	EvalInt(scope *tick.Scope) (int64, error)
+	EvalString(scope *tick.Scope) (string, error)
+	EvalBool(scope *tick.Scope) (bool, error)
+
+	EvalNum(scope *tick.Scope) (interface{}, error)
+}
+
+type StatefulExpr struct {
```

Comment:
There won't be any implement to this interface if he will be private, and we don't need an interface for tests.
So we are "privateizing" it, we can remove it, but we didn't ;)

And you don't need to say sorry for requesting me to change things, it is acceptable!